crates/server/src/authority/catalog.rs

```diff
@@ -836,6 +839,48 @@ async fn build_forwarded_response(
                 (Answer::Normal(Box::new(EmptyLookup)), authorities)
             }
         }
+        #[cfg(all(feature = "__dnssec", feature = "recursor"))]
+        Err(LookupError::RecursiveError(e)) => match e.kind() {
+            ErrorKind::Proto(e) => match e.kind() {
+                ProtoErrorKind::Nsec {
+                    response, proof, ..
+                } if proof.is_insecure() => {
+                    response_header.set_response_code(response.response_code());
+
+                    if let Some(soa) = response.soa() {
+                        let soa = soa.to_owned().into_record_of_rdata();
+                        let record_set = Arc::new(RecordSet::from(soa));
+                        let records = LookupRecords::new(LookupOptions::default(), record_set);
+
+                        (
+                            Answer::NoRecords(Box::new(AuthLookup::SOA(records))),
+                            Box::<AuthLookup>::default(),
+                        )
+                    } else {
+                        (
+                            Answer::Normal(Box::new(EmptyLookup)),
+                            Box::<AuthLookup>::default(),
+                        )
+                    }
+                }
+                _ => {
+                    response_header.set_response_code(ResponseCode::ServFail);
+                    debug!("error resolving {e:?}");
+                    (
+                        Answer::Normal(Box::new(EmptyLookup)),
+                        Box::<AuthLookup>::default(),
+                    )
+                }
+            },
+            _ => {
+                response_header.set_response_code(ResponseCode::ServFail);
+                debug!("error resolving {e:?}");
```

Comment:
nit: We can use structured logging here, as done below
```suggestion
                    debug!(error = ?e, "error resolving");
                    (
                        Answer::Normal(Box::new(EmptyLookup)),
                        Box::<AuthLookup>::default(),
                    )
                }
            },
            _ => {
                response_header.set_response_code(ResponseCode::ServFail);
                debug!(error = ?e, "error resolving");
```